Batch test cleanup/reorganization

jasper-schneider opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
<!--
Reorganized the tests for the Batch cmdlets.  This should bring down the CI execution time for the Batch project.
- Converted some scenario tests to unit tests
- Streamlined the scenario tests, removed some duplicate coverage
- General cleanup
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ x ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [ NA ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [ NA ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [ x ] Title of the pull request is clear and informative.
- [ x ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ x ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [ x ] Pull request includes test coverage for the included changes.
- [ x ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ NA ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [ NA ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThrough` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ NA ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ NA ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ NA ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

markcowl commented:
@jasper-schneider This is 40000 lines of removed test coverage - why is this coverage no longer necessary?

There are no technical issues with the change, but are these tests unnecessary or redundant in some way?

jasper-schneider commented:
@markcowl  
Looks like I messed up copy/pasting my description into the box.  Most of the 40000 lines are from the JSON session records.
There were 3 main things I did during this cleanup effort:
1. I moved a bunch of scenario tests into unit tests.  The unit tests were able to validate the same things, with the bonus of running faster and requiring fewer lines of code.
2. There was some duplicate coverage between some unit tests and scenario tests.  Basically, I wrote some scenario tests first for our OData helpers, then some time later I realized I could/should just write unit tests for those, but I didn't delete the extra scenario tests at that point.  In this PR, I went through and cleaned out the redundant scenario tests.
3. The scenario tests were originally focused on single cmdlets.  So there was a single test for the New cmdlet, followed by a single test for the Get cmdlet, followed by a single test for Set, followed by a single test for Delete.  I talked to Hovsep a while ago about strategies for reducing the test duration for the CI job, and one thing he mentioned was that the goal was to target more end to end basic scenarios in these tests, saving the exhaustive coverage for unit tests.  So I went through and streamlined 4 individual CRUD tests into a single end to end scenario.  Multiply this across our different nouns and all the recorder files that get generated, and that's where the bulk of the lines of code got cut out.

So in short, this was an effort to clean up a bunch of stuff that's accumulated over time, not to create major holes in test coverage :)

Feel free to ping me if you want to discuss in more detail.

markcowl commented:
@jasper-schneider Thanks, just wanted to verify.

on demand run here:http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1314/ 

LGTM once the on-demand run passes
